### PR TITLE
bun: 1.0.0 -> 1.0.1

### PR DESCRIPTION
**Pull request by Abdiramen**
## Description of changes

Bump Bun version from 1.0.0 to 1.0.1 to capture latest bug fixes. [Changelog](https://bun.sh/blog/bun-v1.0.1)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by natsukium — CHANGES_REQUESTED** [review]
The hash is wrong except for x86_64-linux.

```
trying https://github.com/oven-sh/bun/releases/download/bun-v1.0.1/bun-linux-aarch64.zip
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
100 31.6M  100 31.6M    0     0  60.7M      0 --:--:-- --:--:-- --:--:-- 60.7M
error: hash mismatch in fixed-output derivation '/nix/store/594h35lm38xi4wfzbc9g2rjx9p6vvxh9-bun-linux-aarch64.zip.drv':
         specified: sha256-e9YRMyBev3rLDEGJVp5LfTVa8kfHAuBKY0k7a2EDkmw=
            got:    sha256-EEcKoaaC0KnaAfUqd5QNYpTeArY3ia38l5dkZCooZuw=
error: 1 dependencies of derivation '/nix/store/0w3axpwj0hlvcmad1j6x16w6s1r3sqnw-bun-1.0.1.drv' failed to build
```

**Comment by Abdiramen** [comment]
> The hash is wrong except for x86_64-linux.
> 
> 
> 
> ```
> 
> trying https://github.com/oven-sh/bun/releases/download/bun-v1.0.1/bun-linux-aarch64.zip
> 
>   % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
> 
>                                  Dload  Upload   Total   Spent    Left  Speed
> 
>   0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0
> 
> 100 31.6M  100 31.6M    0     0  60.7M      0 --:--:-- --:--:-- --:--:-- 60.7M
> 
> error: hash mismatch in fixed-output derivation '/nix/store/594h35lm38xi4wfzbc9g2rjx9p6vvxh9-bun-linux-aarch64.zip.drv':
> 
>          specified: sha256-e9YRMyBev3rLDEGJVp5LfTVa8kfHAuBKY0k7a2EDkmw=
> 
>             got:    sha256-EEcKoaaC0KnaAfUqd5QNYpTeArY3ia38l5dkZCooZuw=
> 
> error: 1 dependencies of derivation '/nix/store/0w3axpwj0hlvcmad1j6x16w6s1r3sqnw-bun-1.0.1.drv' failed to build
> 
> ```

Hmmm, strange, i'm not sure why i'm getting the wrong hash. I'll dig into this in the morning. Thanks for the review!

**Comment by mfrw** [comment]
Result of `nixpkgs-review pr 254873` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>bun</li>
  </ul>
</details>

File: pkgs/development/web/bun/default.nix [hunk 1]
```diff
@@ -35,19 +35,19 @@ stdenvNoCC.mkDerivation rec {
     sources = {
       "aarch64-darwin" = fetchurl {
         url = "https://github.com/oven-sh/bun/releases/download/bun-v${version}/bun-darwin-aarch64.zip";
-        hash = "sha256-Bd8KL1IbWBRiMZq4YPhNLdhBOqRReCFeUPAilLfk0TM=";
+        hash = "sha256-6qSq/mR+f8wlrMwHqNEZhStULISmVidetR3LGwPYfv8=";
```
Comment (cdmistman) [inline pkgs/development/web/bun/default.nix:38]:
```suggestion
        hash = "sha256-AzjCqfvR4Uad6+ocLo+KfqHKFjxOlZt4//Nw80FRijM=";
```

File: pkgs/development/web/bun/default.nix [hunk 2]
```diff
@@ -35,19 +35,19 @@ stdenvNoCC.mkDerivation rec {
...
       };
       "aarch64-linux" = fetchurl {
         url = "https://github.com/oven-sh/bun/releases/download/bun-v${version}/bun-linux-aarch64.zip";
-        hash = "sha256-CHOiQ47wXjkFyJG9ElE9gBpmWpylMEUf6c+Sm+YCpGc=";
+        hash = "sha256-e9YRMyBev3rLDEGJVp5LfTVa8kfHAuBKY0k7a2EDkmw=";
```
Comment (cdmistman) [inline pkgs/development/web/bun/default.nix:42]:
```suggestion
        hash = "sha256-EEcKoaaC0KnaAfUqd5QNYpTeArY3ia38l5dkZCooZuw=";
```

File: pkgs/development/web/bun/default.nix [hunk 3]
```diff
@@ -35,19 +35,19 @@ stdenvNoCC.mkDerivation rec {
...
       };
       "x86_64-darwin" = fetchurl {
         url = "https://github.com/oven-sh/bun/releases/download/bun-v${version}/bun-darwin-x64.zip";
-        hash = "sha256-0AT58hjawS60q5YAQd/upVz0vOIs11JM+lc3c1mGyOE=";
+        hash = "sha256-vR2+4kuTlzowGTarQZj2WEpINxzoo55Ygg0vKjX1rPs=";
```
Comment (cdmistman) [inline pkgs/development/web/bun/default.nix:46]:
```suggestion
        hash = "sha256-g4k0NdpWU1K+66Mi/idYAMx8pFLOBbUDk4rVczrxRjg=";
```

**Review by cdmistman — COMMENTED** [review]
got the correct hashes :) please be sure to collapse into 1 commit!

File: pkgs/development/web/bun/default.nix (hunk 3 above)
Comment (Abdiramen) [reply pkgs/development/web/bun/default.nix]: updated here: https://github.com/NixOS/nixpkgs/pull/254873/commits/a57315eed02fda5e5c0a3139fe6ab5beb438a11f

File: pkgs/development/web/bun/default.nix (hunk 1 above)
Comment (Abdiramen) [reply pkgs/development/web/bun/default.nix]: updated here: https://github.com/NixOS/nixpkgs/pull/254873/commits/a57315eed02fda5e5c0a3139fe6ab5beb438a11f

**Comment by Abdiramen** [comment]
> got the correct hashes :) please be sure to collapse into 1 commit!

Thanks Colton!

**Comment by cdmistman** [comment]
of course! your PR should still be 2 commits though: `maintainers: add abdiramen` and `bun: 1.0.0 -> 1.0.1`. please squash the 3rd commit into the 2nd commit

**Review by cdmistman — APPROVED** [review]

**Review by cdmistman — APPROVED** [review]

**Review by natsukium — APPROVED** [review]
LGTM, thanks.

**Pull request merged**